Lesson4/src/com/home/model/Person.java

```diff
@@ -86,4 +87,16 @@ public void setGender(char gender) {
         this.gender = gender;
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Person person = (Person) o;
+        return gender == person.gender && age == person.age && height == person.height && address.equals(person.address) && name.equals(person.name);
```

Comment:
this is good, but note that if the address or name is not initialized (value is null) -- you will get an error here